### PR TITLE
support access control for job

### PR DESCRIPTION
**Pull request by tornado404**
### PR types
add PR template

### PR changes
Others

### Describe
support access control for job

### CODE REVIEW

**Review by luoyuedong — APPROVED** [review]

**Review by HaozhengAN — APPROVED** [review]

File: pkg/apiserver/controller/job/create_test.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
+package job
+
+import (
+	"github.com/PaddlePaddle/PaddleFlow/pkg/common/config"
```
Comment (D0m021ng) [inline pkg/apiserver/controller/job/create_test.go:8]: fmt

File: pkg/apiserver/controller/job/create_test.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+					},
+					Framework: schema.FrameworkStandalone,
+				},
+			},
+			wantErr:      false,
+			responseCode: 400,
+		},
+	}
+
+	for _, tt := range tests {
```
Comment (D0m021ng) [inline pkg/apiserver/controller/job/create_test.go:71]: Add `t.Run` in for loop
```go
t.Run(tt.name, func(t *testing.T) {

})```

File: pkg/apiserver/controller/job/job_test.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+import (
+	"github.com/PaddlePaddle/PaddleFlow/pkg/common/config"
+	"github.com/PaddlePaddle/PaddleFlow/pkg/common/logger"
+	"github.com/PaddlePaddle/PaddleFlow/pkg/common/resources"
+	"github.com/PaddlePaddle/PaddleFlow/pkg/common/schema"
+	"github.com/PaddlePaddle/PaddleFlow/pkg/model"
+	"github.com/PaddlePaddle/PaddleFlow/pkg/storage"
+	"github.com/PaddlePaddle/PaddleFlow/pkg/storage/driver"
+	"github.com/stretchr/testify/assert"
```
Comment (D0m021ng) [inline pkg/apiserver/controller/job/job_test.go]: format

File: pkg/apiserver/controller/job/job_test.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	"testing"
+)
+
+const (
+//mockRootUser = "root"
```
Comment (D0m021ng) [inline pkg/apiserver/controller/job/job_test.go]: please remove this.

File: pkg/apiserver/controller/job/job_test.go [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+					UserName: mockRootUser,
+				},
+				req: "empty",
+			},
+			wantErr:      true,
+			responseCode: 400,
+		},
+	}
+
+	for _, tt := range tests {
```
Comment (D0m021ng) [inline pkg/apiserver/controller/job/job_test.go:65]: Add `t.Run`

File: pkg/apiserver/router/v1/job_test.go [hunk 6]
```diff
@@ -125,3 +145,216 @@ func TestCreateJob(t *testing.T) {
...
+					JobID:    MockJobID,
+					Priority: schema.EnvJobHighPriority,
+				},
+				router: routerNonRoot,
+			},
+			wantErr:      true,
+			responseCode: 403,
+		},
+	}
+	for _, tt := range tests {
```
Comment (D0m021ng) [inline pkg/apiserver/router/v1/job_test.go:223]: Add `t.Run`

File: pkg/apiserver/router/v1/job_test.go [hunk 7]
```diff
@@ -125,3 +145,216 @@ func TestCreateJob(t *testing.T) {
...
+			args: args{
+				ctx:    &logger.RequestContext{UserName: MockNonRootUser2},
+				req:    MockJobID,
+				router: router,
+			},
+			wantErr:      false,
+			responseCode: 200,
+		},
+	}
+	for _, tt := range tests {
```
Comment (D0m021ng) [inline pkg/apiserver/router/v1/job_test.go:285]: Add `t.Run`

File: pkg/apiserver/router/v1/job_test.go [hunk 8]
```diff
@@ -125,3 +145,216 @@ func TestCreateJob(t *testing.T) {
...
```
Comment (D0m021ng) [inline pkg/apiserver/router/v1/job_test.go:349]: Add `t.Run`

File: pkg/apiserver/controller/job/create_test.go (hunk 2 above)
Comment (tornado404) [reply pkg/apiserver/controller/job/create_test.go:71]: done

File: pkg/apiserver/controller/job/job_test.go (hunk 5 above)
Comment (tornado404) [reply pkg/apiserver/controller/job/job_test.go:65]: done

File: pkg/apiserver/router/v1/job_test.go (hunk 8 above)
Comment (tornado404) [reply pkg/apiserver/router/v1/job_test.go:349]: done

File: pkg/apiserver/controller/job/job_test.go [hunk 9]
```diff
@@ -0,0 +1,76 @@
+package job
+
+import (
+	"github.com/stretchr/testify/assert"
```
Comment (D0m021ng) [inline pkg/apiserver/controller/job/job_test.go]: format

**Review by D0m021ng — APPROVED** [review]

**Pull request merged**